shared/dataflow/codeql/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -895,12 +1021,24 @@ module Make<
       } or
       TSummaryParameterNode(SummarizedCallable c, ParameterPosition pos) {
         summaryParameterNodeRange(c, pos)
+      } or
+      TSourceOutputNode(SourceNode source, SummaryNodeState state, string kind, string model) {
+        state.isSourceOutputState(source, _, kind, model)
+      } or
+      TSinkInputNode(SinkNode sink, SummaryNodeState state, string kind, string model) {
+        state.isSinkInputState(sink, _, kind, model)
```

Comment:
I think something needs to change here. We worked very hard to untangle and stratify `DataFlow::Node` vs. `SummaryNode` such that `FlowSummaryImpl` wasn't dependent on `DataFlow::Node`. But now they've been made mutually recursive. We need to fix that, i.e. we should not embed `DataFlow::Node`s into `SummaryNode`s.

Reply:
`SinkNode` is not a subset of `DataFlow::Node`, rather it is a subset of the parameterized type `SinkBase`, which for Rust is subset of `AstNode`.

Reply:
Ah! Right.
So I guess the only new connection to `DataFlow::Node` is in `sourceLocalStep` via `StepsInput::getSourceNode`?

Reply:
Maybe `SourceNode` and `SinkNode` ought to be called something different, because it's perhaps somewhat unclear what kind of "nodes" they are supposed to be - at least I got confused. They're not `DataFlow::Node`s and they are not `SummaryNode`s. Maybe simply use `SourceElement`/`SinkElement` instead? And likely elaborate related qldoc a bit.